README.md

```diff
@@ -149,7 +149,44 @@ final connection = InternetConnection.createInstance(
 );
 ```
 
-### 7. Pause and Resume on App Lifecycle Changes
+### 7. Using a custom connectivity check method
+
+For advanced use cases, you can completely customize how connectivity checks are performed by providing your own connectivity checker:
+
+```dart
+final connection = InternetConnection.createInstance(
+  customConnectivityCheck: (option) async {
+    // Example: Use a different HTTP method or client
+    try {
+      final client = http.Client();
+      try {
+        final response = await client.get(
+          option.uri,
+          headers: option.headers,
+        ).timeout(option.timeout);
+        
+        return InternetCheckResult(
+          option: option,
+          isSuccess: response.statusCode == 200,
+        );
+      } finally {
+        client.close();
+      }
+    } catch (_) {
+      return InternetCheckResult(option: option, isSuccess: false);
+    }
+  },
```

Comment:
This is kinda what the default integration does. Can you add a more complex / advance implementation so that developers can understand how exactly this option works / should be used.